src/thriveopedia/pages/ThriveopediaEvolutionaryTreePage.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Godot;
+
+/// <summary>
+///    Thriveopedia page displaying an evolutionary tree of all species in the current game.
+/// </summary>
+/// <remarks>
+///   <para>
+///     Note a lot of this functionality is duplicated from AutoEvoExploringTool.
+///   </para>
+/// </remarks>
+public class ThriveopediaEvolutionaryTreePage : ThriveopediaPage
+{
+    [Export]
+    public NodePath DisabledInFreebuildPath = null!;
+
+    [Export]
+    public NodePath EvolutionaryTreePath = null!;
+
+    [Export]
+    public NodePath SpeciesDetailsLabelPath = null!;
+
+    [Export]
+    public NodePath SpeciesPreviewPath = null!;
+
+    [Export]
+    public NodePath HexPreviewPath = null!;
+
+    private readonly List<Dictionary<uint, Species>> speciesHistoryList = new();
+    private VBoxContainer disabledInFreebuild = null!;
+    private EvolutionaryTree evolutionaryTree = null!;
+    private CustomRichTextLabel speciesDetailsLabel = null!;
+    private SpeciesPreview speciesPreview = null!;
+    private CellHexesPreview hexesPreview = null!;
+
+    public override string PageName => "EvolutionaryTree";
+    public override string TranslatedPageName => TranslationServer.Translate("EVOLUTIONARY_TREE_PAGE");
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        disabledInFreebuild = GetNode<VBoxContainer>(DisabledInFreebuildPath);
+        evolutionaryTree = GetNode<EvolutionaryTree>(EvolutionaryTreePath);
+        speciesDetailsLabel = GetNode<CustomRichTextLabel>(SpeciesDetailsLabelPath);
+        speciesPreview = GetNode<SpeciesPreview>(SpeciesPreviewPath);
+        hexesPreview = GetNode<CellHexesPreview>(HexPreviewPath);
+
+        UpdateCurrentWorldDetails();
+    }
+
+    public override void UpdateCurrentWorldDetails()
+    {
+        if (CurrentGame == null)
+            return;
+
+        RebuildTree();
+    }
+
+    public override void OnNavigationPanelSizeChanged(bool collapsed)
+    {
+    }
+
+    /// <summary>
+    ///   Clear and then rebuild the evolutionary tree each time we open the page. This way, we ensure the tree is
+    ///   always up to date.
+    /// </summary>
+    public void RebuildTree()
+    {
+        if (!Visible)
+            return;
+
+        // TODO fix the tree for freebuild?
+        if (CurrentGame!.FreeBuild)
```

Comment:
I'd say do the longer thing and put a `if(CurrentGame == null) throw new invalidoperationexception` at the top of this method instead of just skip over it like this. I've done that in quite many places in the game.